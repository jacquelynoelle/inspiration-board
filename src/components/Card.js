import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  generateEmoji = () => {
    const { cardEmoji } = this.props

    return cardEmoji && emoji.getUnicode(cardEmoji);
  }

  deleteCard = () => {
    this.props.deleteCardCallback(this.props.id);
  }

  render() {
    return (
      <div className="card">
        <div className="card__content">
          <span className="card__content-text">{ this.props.cardText }</span>
          <span className="card__content-emoji">{ this.generateEmoji() }</span>
        </div>
        <span onClick={ this.deleteCard } className="card__delete">X</span>
      </div>
    )
  }
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  cardText: PropTypes.string,
  cardEmoji: PropTypes.string,
  deleteCardCallback: PropTypes.func.isRequired,
};

export default Card;
