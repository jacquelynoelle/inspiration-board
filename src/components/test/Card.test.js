import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // Arrange
    const wrapper = shallow( <Card addCardCallback={ () => {} } />);

    // Assert
    expect(wrapper).toMatchSnapshot();
  });
});
