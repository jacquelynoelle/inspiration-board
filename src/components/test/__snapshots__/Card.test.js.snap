// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card__content"
        >
          <span
            className="card__content-text"
          />
          <span
            className="card__content-emoji"
          />
        </div>,
        <span
          className="card__delete"
          onClick={[Function]}
        >
          X
        </span>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card__content-text"
            />,
            <span
              className="card__content-emoji"
            />,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card__delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card__content"
          >
            <span
              className="card__content-text"
            />
            <span
              className="card__content-emoji"
            />
          </div>,
          <span
            className="card__delete"
            onClick={[Function]}
          >
            X
          </span>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card__content-text"
              />,
              <span
                className="card__content-emoji"
              />,
            ],
            "className": "card__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card__content-text",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card__content-emoji",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card__delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
